### PR TITLE
feat: add codspeed for continuous benchmarking

### PR DESCRIPTION
**Pull request by erikwrede**
This PR adds CodSpeed for benchmarking

### CODE REVIEW

File: tests/benchmarks/test_visit.py [hunk 1]
```diff
@@ -23,5 +23,5 @@ def test_visit_all_ast_nodes(benchmark, big_schema_sdl):  # noqa: F811
 def test_visit_all_ast_nodes_in_parallel(benchmark, big_schema_sdl):  # noqa: F811
     document_ast = parse(big_schema_sdl)
     visitor = DummyVisitor()
-    parallel_visitor = ParallelVisitor([visitor] * 50)
+    parallel_visitor = ParallelVisitor([visitor] * 20)
```
Comment (Cito) [inline tests/benchmarks/test_visit.py:26]: Is there a reason why you reduced the number? The original benchmark had 50.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -3,6 +3,29 @@ name: Tests
...
 
 jobs:
+
+  benchmarks:
+    name: 📈 Benchmarks
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - run: pipx install poetry
```
Comment (Cito) [inline .github/workflows/test.yml]: Shouldn't this come after the setup-python?

**Review by Cito — APPROVED** [review]
Thanks, looks good. I have made two comments. I'd also move it into a separate yml file.

You can ignore the mypy issues since they are not related to the PR, I fill fix them in main.

File: tests/benchmarks/test_visit.py (hunk 1 above)
Comment (erikwrede) [reply tests/benchmarks/test_visit.py:26]: Benchmark timed out on CI with 50 visitors. Couldn't reproduce, so I reduced the number and now it's working perfectly fine.

File: .github/workflows/test.yml (hunk 2 above)
Comment (erikwrede) [reply .github/workflows/test.yml]: Good point, I honestly copied this job over from strawberry, where' it's been running smoothly for a while. I can change the order if you prefer.

File: tests/benchmarks/test_visit.py (hunk 1 above)
Comment (Cito) [reply tests/benchmarks/test_visit.py:26]: Ok, just wanted to know. I'll add a comment.

File: .github/workflows/test.yml (hunk 2 above)
Comment (Cito) [reply .github/workflows/test.yml]: Saw you did that already, thanks.

**Pull request merged**